integrations/slack/save.go

```diff
@@ -0,0 +1,70 @@
+package slack
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/integrations/slack/internal/vars"
+	"go.autokitteh.dev/autokitteh/sdk/sdkintegrations"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+	"go.uber.org/zap"
+)
+
+const (
+	headerContentType = "Content-Type"
+	contentTypeForm   = "application/x-www-form-urlencoded"
+)
+
+func (h handler) handleSave(w http.ResponseWriter, r *http.Request) {
+	c, l := sdkintegrations.NewConnectionInit(h.logger, w, r, desc)
+
+	// Check "Content-Type" header.
+	contentType := r.Header.Get(headerContentType)
+	if r.Method == http.MethodPost && !strings.HasPrefix(contentType, contentTypeForm) {
+		c.AbortBadRequest("unexpected content type")
+		return
+	}
+
+	// Read and parse POST request body.
+	if err := r.ParseForm(); err != nil {
+		l.Warn("Failed to parse incoming HTTP request", zap.Error(err))
+		c.AbortBadRequest("form parsing error")
+		return
+	}
+
+	clientID := r.FormValue("client_id")
+	clientSecret := r.FormValue("client_secret")
+	signingSecret := r.FormValue("signing_secret")
```

Comment:
Validate they are not empty, maybe other limitations?

Reply:
No need. This is an (uncommon) case where clide-side checks are sufficient. If a user sends a request manually, and it's malformed, that's their problem because we fail safely anyway.